pkg/process/checks/net.go

```diff
@@ -85,12 +90,102 @@ func (c *ConnectionsCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.
 	// Resolve the Raddr side of connections for local containers
 	LocalResolver.Resolve(conns)
 
+	// Get the container and process relationship from either the process or container checks
+	ctrsByPid := ctrIDsByPIDs(connectionPIDs(conns.Conns))
+
+	resolveLoopbackConnections(conns.Conns, ctrsByPid)
+
 	tel := c.diffTelemetry(conns.Telemetry)
 
 	log.Debugf("collected connections in %s", time.Since(start))
 	return batchConnections(cfg, groupID, c.enrichConnections(conns.Conns), conns.Dns, c.networkID, tel), nil
 }
 
+func resolveLoopbackConnections(conns []*model.Connection, ctrsByPid map[int32]string) {
```

Comment:
Thoughts on moving this logic somewhere within `pkg/network` and calling out to it from here? This network check is getting a little unwieldy

Reply:
I noticed we used `LocalResolver.LoadAddrs(ctrList)` below - sounds like this might be a good fit to live within the `LocalResolver`?

Reply:
Yes, good idea, done.